js-arrays/arrays.js

```diff
@@ -0,0 +1,202 @@
+function isArrayOne(a) {
+  return a instanceof Array;
+}
+
+function isArrayTwo(a) {
+  return Array.isArray(a);
+}
+
+function isArrayThree(a) {
+  return Object.prototype.toString.call(a) === "[object Array]";
+}
+
+function range(a, b, c) {
+  var arr = [];
+  var i = 0;
+  var j, k;
+  var diff = b - a;
+  if (b == undefined) {
+    j = a;
+  } else {
+    i = a;
+    j = b;
+  }
+  if (c == undefined || c == 0) {
+    if (diff < 0) {
+      return arr;
+    } else {
+      k = 1;
+    }
+  } else if (c > 0) {
+    if (diff < 0) {
+      return arr;
+    } else {
+      k = c;
+      for (i; i < j; i += k) {
+        arr.push(i);
+      }
+    }
+  } else if (c < 0) {
+    if (diff < 0) {
+      k = c;
+      for (i; i > j; i += k) {
+        arr.push(i);
+      }
+    }
+  }
+
+  return arr;
+}
+
+function compactLoop(a) {
+  var arr = [];
+  if (Array.isArray(a)) {
+    for (var i = 0; i < a.length; i++) {
+      if (a[i]) {
+        arr.push(a[i]);
+      }
+    }
+    return arr;
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function compact(a) {
+  if (Array.isArray(a)) {
+    return a.filter(function(el) {
+      return el ? true : false; 
+    });
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function sumLoop(a) {
+  if (Array.isArray(a)) {
+    var sum = 0;
+    for (var i = 0; i < a.length; i++) {
+      sum += a[i];
+    }
+    return sum;
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function sum(a) {
+  if (Array.isArray(a)) {
+    return a.reduce(function(a,b) {
+      return a + b;
+    });
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function unique(a) {
+  if (Array.isArray(a)) {
+    var arr = [];
+    a.forEach(function(el) {
+      if (!arr.includes(el)) {
+        arr.push(el);
+      }
+    });
+    return arr;
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function last(a) {
+  if (Array.isArray(a)) {
+    return a[a.length-1];
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function excludeLast(a, b) {
+  if (Array.isArray(a)) {
+    if (b === undefined) {
+      return a.slice(0, a.length - 1);
+    } else if (typeof b == "number") {
```

Comment:
it's JS, no need to check type
what do you think about `var n = n || 1;  return a.slice(0, a.length - n); `?